chia/util/merkle_set.py

```diff
@@ -120,18 +120,17 @@ def _audit(self, hashes: List[bytes], bits: List[int]) -> None:
 class MerkleSet:
     root: Node
 
-    def __init__(self, root: Optional[Node] = None):
-        if root is None:
-            self.root = _empty
+    def __init__(self, leafs: List[bytes32], *, set_root: Optional[Node] = None):
```

Comment:
The mutually exclusively used parameters seems a bit fishy vs. having two constructors.  Also, when passing `set_root` you are still required to pass `leafs` despite it being unused.

Maybe both should be optional and then use `@typing.overload` to express the appropriate modes of use.  Or just have two (or three) separate ways to create a set since there is no actually shared code here.

Reply:
yeah, the `set_root` alternative is internal as well, I'll make that a separate constructor function